lib/internal/util/inspect.js

```diff
@@ -1248,7 +1248,7 @@ function getClassBase(value, constructor, tag) {
   if (constructor !== 'Function' && constructor !== null) {
     base += ` [${constructor}]`;
   }
-  if (tag !== '' && constructor !== tag) {
+  if (tag !== '' && (constructor === null || !constructor.includes(tag))) {
```

Comment:
I think we can keep the former check in this case and also in the one below for functions. The reason is, that extending classes already has no duplication by default and adding the tag on these is good to know about.
I am also not sure which of the new tests would fail by removing this, just looking at the tests.

```suggestion
  if (tag !== '' && constructor !== tag) {
```

Reply:
Oh thank you for pointing that! I'll fix both